control/autoware_lane_departure_checker/src/lane_departure_checker_node/utils.cpp

```diff
@@ -163,6 +165,64 @@ std::vector<LinearRing2d> createVehicleFootprints(
   return vehicle_footprints;
 }
 
+lanelet::ConstLanelets getCandidateLanelets(
+  const lanelet::ConstLanelets & route_lanelets,
+  const std::vector<LinearRing2d> & vehicle_footprints)
+{
+  lanelet::ConstLanelets candidate_lanelets;
+
+  // Find lanes within the convex hull of footprints
+  const auto footprint_hull = createHullFromFootprints(vehicle_footprints);
+
+  for (const auto & route_lanelet : route_lanelets) {
+    const auto poly = route_lanelet.polygon2d().basicPolygon();
+    if (!boost::geometry::disjoint(poly, footprint_hull)) {
+      candidate_lanelets.push_back(route_lanelet);
+    }
+  }
+
+  return candidate_lanelets;
+}
+
+LinearRing2d createHullFromFootprints(const std::vector<LinearRing2d> & footprints)
+{
+  MultiPoint2d combined;
+  for (const auto & footprint : footprints) {
+    for (const auto & p : footprint) {
+      combined.push_back(p);
+    }
+  }
+
+  LinearRing2d hull;
+  boost::geometry::convex_hull(combined, hull);
+
+  return hull;
+}
+
+std::vector<LinearRing2d> createVehiclePassingAreas(
+  const std::vector<LinearRing2d> & vehicle_footprints)
+{
+  if (vehicle_footprints.empty()) {
+    return std::vector<LinearRing2d>();
+  }
+
+  std::vector<LinearRing2d> areas;
+  areas.reserve(vehicle_footprints.size() == 1 ? 1 : vehicle_footprints.size() - 1);
+
+  if (vehicle_footprints.size() == 1) {
+    areas.push_back(vehicle_footprints.front());
+    return areas;
+  }
```

Comment:
```suggestion
  if (vehicle_footprints.size() == 1) {
    return {vehicle_footprints.front()};
  }
  
  std::vector<LinearRing2d> areas;
  areas.reserve(vehicle_footprints.size() - 1);
```
Since this is not refactoring PR, so it's fine not to commit this.
But in case you're refactoring, maybe you can do something like this to avoid having to add logic in the reserve.